src/Device/Driver/FreeVario.cpp

```diff
@@ -281,6 +281,35 @@ FreeVarioDevice::OnCalculatedUpdate(const MoreData &basic,
 
   char nmeaOutbuffer[80];
 
+ if (basic.external_wind_available.IsValid() && basic.track_available){
+   int track = basic.track.Degrees();
```

Comment:
Save yourself the conversion to int, and do this directly in the relWindDirection below.
True for all other instances aswell.

Reply:
Sorry, then I misunderstood you. I thought that was a request from you to change that! I have now eliminated the integers and also added a query for the calculated wind as to whether the external wind is definitely invalid.